### PR TITLE
Dataflow Cloud Code Pub/Sub to BigQuery sample

### PR DESCRIPTION
**Pull request by damondouglas**
This PR implements a Dataflow Cloud Code plugin dependency providing:

- an Apache Beam pipeline example that reads from Pub/Sub and writes to BigQuery
- Infrastructure-as-Code (IaC) to provision dependent Google Cloud resources

Some notes to the reviewer:

1) The README still contains placeholders for the Getting Started with IntelliJ and Sign up for User Research sections.
2) The Open In Cloud Shell button references the v1 branch of this repository and obviously will not work.  Instead use the following URL: https://shell.cloud.google.com/cloudshell/editor?cloudshell_git_repo=https%3A%2F%2Fgithub.com%2Fdamondouglas%2Fcloud-code-samples&cloudshell_git_branch=read-pubsub-write-bigquery&cloudshell_tutorial=cloud-shell-readme.md&cloudshell_working_dir=java%2Fjava-dataflow-samples%2Fread-pubsub-write-bigquery&show=terminal

### CODE REVIEW

File: java/java-dataflow-samples/read-pubsub-write-bigquery/README.md
Comment (ivanporty) [inline java/java-dataflow-samples/read-pubsub-write-bigquery/README.md]: if we want to keep this sample specifically for Cloud Code, we should assume Cloud SDK is installed already by Cloud Code and init is done. Auth could be done using Cloud Code menu item so no CLI work needed.

File: .templates/template_location_v2.json
Comment (peterlin741) [inline .templates/template_location_v2.json]: nit: I see that `read-pubsub-write-bigquery` is repeated, in the `directoryPath` and `templatePath`. Just wanted to make sure this naming was intentional, since the other examples have different naming between the fields

Comment (ivanporty) [reply .templates/template_location_v2.json]: this is only used in metrics as a template type.
for CC samples we have "hello-world" and "guestbook"
for dataflow you can pick any meaningful name here

File: .templates/template_location_v2.json
Comment (damondouglas) [inline .templates/template_location_v2.json:185]: @peterlin741 / @ivanporty would this work?

File: java/java-dataflow-samples/read-pubsub-write-bigquery/src/main/java/com/cloudcode/dataflow/example/logging/Log.java
Comment (damondouglas) [inline java/java-dataflow-samples/read-pubsub-write-bigquery/src/main/java/com/cloudcode/dataflow/example/logging/Log.java:72]: @damondouglas add return

File: java/java-dataflow-samples/read-pubsub-write-bigquery/cloud-shell-readme.md
Comment (damondouglas) [inline java/java-dataflow-samples/read-pubsub-write-bigquery/cloud-shell-readme.md]: @damondouglas "Move this file to your IDE working directory on your local machine"

File: .templates/template_location_v2.json
Comment (ivanporty) [reply .templates/template_location_v2.json:185]: looks good!

File: java/java-dataflow-samples/read-pubsub-write-bigquery/.readmes/intellij/README.md
Comment (ivanporty) [inline java/java-dataflow-samples/read-pubsub-write-bigquery/.readmes/intellij/README.md:61]: do we want to fill this in later once implementation completes?

Comment (damondouglas) [reply java/java-dataflow-samples/read-pubsub-write-bigquery/.readmes/intellij/README.md:61]: @ivanporty I'd vote yes for filling in later as we may discover other hopefully small changes.

**Review by ivanporty — APPROVED** [review]
Thank you, LGTM. I'll try to find who can give us a code owner approval to speed things up

**Comment by damondouglas** [comment]
@ivanporty I noticed that files in this repository do not have license headers.  Will we need to add license headers to all the files in the future or assume coverage from https://github.com/GoogleCloudPlatform/cloud-code-samples/blob/v1/LICENSE?

**Comment by ivanporty** [comment]
> @ivanporty I noticed that files in this repository do not have license headers. Will we need to add license headers to all the files in the future or assume coverage from https://github.com/GoogleCloudPlatform/cloud-code-samples/blob/v1/LICENSE?

Good question, I will defer this to repo owners tagged for approvals.

**Comment by ivanporty** [comment]
@iennae @averikitsch Could you please quickly glance at this PR and approve to unblock our dataflow sample integration. Thanks!

**Comment by averikitsch** [comment]
This is the wrong repository to put this sample code. Please work with your DEE team ([go/dee-data-ai](https://goto.google.com/dee-data-ai)) to identify the correct repository and sample guidelines.

**Comment by ivanporty** [comment]
@averikitsch this is a sample code for Cloud Code IntelliJ, we do have samples for K8s, Cloud Run, and Functions here in the repo, what makes this different, could you please elaborate a bit on this?

**Comment by gennadiycivil** [comment]
@kschaab , courtesy ping , do you have any concerns on this PR from VsCode point of view? This is currently blocking DataFlow integration progress on IJ side, should have no effect on VsCode.

**Review by averikitsch — APPROVED** [review]
The gradle-wrapper.jar will cause a security policy violation ([example](https://github.com/GoogleCloudPlatform/cloud-code-samples/issues/1193)). Can this be removed?

**Review by ivanporty — APPROVED** [review]
Approved, @averikitsch please merge if this looks good, thank you!

**Pull request merged**
